Documentation/backends.md

```diff
@@ -29,6 +29,30 @@ host-gw provides good performance, with few dependencies, and easy set up.
 Type:
 * `Type` (string): `host-gw`
 
+### WireGuard
+
+Use in-kernel [WireGuard](https://www.wireguard.com) to encapsulate and encrypt the packets.
+
+Type:
+* `Type` (string): `wireguard`
+* `PSK` (string): Optional. The pre shared key to use. Use `wg genpsk` to generate a key.
+* `ListenPort` (int): Optional. The udp port to listen on. Default is `51820`.
+* `ListenPortV6` (int): Optional. The udp port to listen on for ipv6. Default is `51821`.
+* `Mode` (string): Optional.
+    * separate - Use separate wireguard tunnels for ipv4 and ipv6 (default)
+    * auto - Single wireguard tunnel for both address families; autodetermine the preferred peer address
+    * ipv4 - Single wireguard tunnel for both address families; use ipv4 for
+      the peer addresses
+    * ipv6 - Single wireguard tunnel for both address families; use ipv6 for
+      the peer addresses
+* `PersistentKeepaliveInterval` (int): Optional. Default is 0 (disabled).
+
+If no private key was generated before the private key is written to `/run/flannel/wgkey`. You can use environment `WIREGUARD_KEY_FILE` to change this path.
+
+The static names of the interfaces are `flannel-wg` and `flannel-wg-v6`. WireGuard tools like `wg show` can be used to debug interfaces and peers.
+
+Users of kernels < 5.6 need to [install](https://www.wireguard.com/install/) a module.
```

Comment:
Could you specify the module?

Reply:
Following the URL you should have what you need to install based on your OS. Ubuntu, debian and openSUSE have different packages to install.

Reply:
Aaah ok! I think the word module is confusing. What about modifying the `a module` by `an additional wireguard package`